### ISSUE TITLE
Problem with contact's names or fallbacks

### ISSUE THREAD

**Issue by Lehnemann** [KnugiHK/WhatsApp-Chat-Exporter#58]
When trying to export the backup I receive the following message: 

![image](https://github.com/KnugiHK/WhatsApp-Chat-Exporter/assets/2759337/a906c4e6-0eb5-414d-9fca-206f2c09f885)

`UnboundLocalError: cannot access local variable 'name' where it is not associated with a value`

Please, what can I do about it?

Thanks in advance...

**Comment by KnugiHK** [comment]
It is a bug. Will fix it later.

**Comment by KnugiHK** [comment]
Fixed in bd2f063. Please check if the fix works by using the `dev` branch.

**Comment by Lehnemann** [comment]
Thanks for your work but I dunno how to use it from there... 😐

**Comment by KnugiHK** [comment]
```python
git clone -b dev https://github.com/KnugiHK/WhatsApp-Chat-Exporter
pip install pycryptodome javaobj-py3 ordered-set zstandard nuitka
pip install .
python -m nuitka --onefile --include-data-file=./Whatsapp_Chat_Exporter/whatsapp.html=./Whatsapp_Chat_Exporter/whatsapp.html --assume-yes-for-downloads --follow-imports Whatsapp_Chat_Exporter\__main__.py
```
The above commands will produce a file called `__main__.exe` and that should do the trick.

**Comment by Lehnemann** [comment]
![image](https://github.com/KnugiHK/WhatsApp-Chat-Exporter/assets/2759337/508fc7db-dce7-401a-9ddb-16e69bda4a41)

After the building process, the antivirus blocks the file. The antivirus said that It is a VHO:Worm.Win32.Convagent.gen virus... Is It a false positive?

**Comment by KnugiHK** [comment]
> ![image](https://user-images.githubusercontent.com/2759337/263073203-508fc7db-dce7-401a-9ddb-16e69bda4a41.png)
> 
> After the building process, the antivirus blocks the file. The antivirus said that It is a VHO:Worm.Win32.Convagent.gen virus... Is It a false positive?

Yes. See #43.

**Comment by Lehnemann** [comment]
Well, I tried with the fix and it worked!

Thanks for your help and time! 🙂

You made a great software! Thanks for your effort! 🤩

**Comment by KnugiHK** [comment]
Thanks for your compliment! If you would like to support the project, consider starring this repo.

I will leave this open until the next release.

**Comment by KnugiHK** [comment]
Released in [0.9.7](https://github.com/KnugiHK/WhatsApp-Chat-Exporter/releases/tag/0.9.7).
